### PR TITLE
create html form to submit images via post

### PR DESCRIPTION
**Pull request by rithvikrao**

### CODE REVIEW

File: public/userview.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<html lang="en">
+    <form action="http://localhost:8765" method="post">
```
Comment (akainth015) [inline public/userview.html]: Doing it like this means the user will see just a class after submitting the file. Maybe it is better to do it with an AJAX request and show the server response somewhere.

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

**Review by akainth015 — CHANGES_REQUESTED** [review]

File: public/userview.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+        <label for="img">Select image:</label>
+        <input type="file" id="img" name="img" accept="image/*">
```
Comment (akainth015) [inline public/userview.html]: if you could make a preview feature that would be perfect, [here](https://stackoverflow.com/a/27165977/7267809) is how to do it (use the `FileReader` API)

**Review by akainth015 — CHANGES_REQUESTED** [review]

File: public/userview.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            if (file) {
+                reader.readAsDataURL(file);
+            }
+        }
+    </script>
+
+    <!-- Referenced https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch -->
+    <script>
+        function upload() {
+            document.getElementById("form1").addEventListener("click", function(event){
```
Comment (akainth015) [inline public/userview.html]: Do this outside of `upload`, and use the `submit` event instead of `click`

File: public/userview.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                event.preventDefault()
+            });
+
+            const formData = new FormData();
+            const fileField = document.querySelector('input[type="file"]');
+            formData.append('image', fileField.files[0]);
+
+            fetch('http://localhost:8765', {
+                method: 'PUT',
```
Comment (akainth015) [inline public/userview.html]: Stick to POST, not sure how/if BentoML will handle PUT requests

File: public/userview.html [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    </script>
+
+    <!-- Referenced https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch -->
+    <script>
+        function upload() {
+            document.getElementById("form1").addEventListener("click", function(event){
+                event.preventDefault()
+            });
+
+            const formData = new FormData();
```
Comment (akainth015) [inline public/userview.html]: You can do `new FormData(document.getElementById("form1"))`

File: public/userview.html [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                body: formData
+            })
+            .then(response => response.text())
+            .then(classification => document.getElementById("p1").textContent = classification);
+        }
+    </script>
+
+    <form id="form1" action="javascript:;" onsubmit="upload(this)"></form>
+        <label for="img">Select image:</label>
+        <input type="file" accept="image/*" onchange="showImg()">
```
Comment (akainth015) [inline public/userview.html]: give this `id="img"` so that the label above's `for` works properly

**Review by akainth015 — CHANGES_REQUESTED** [review]

File: public/userview.html (hunk 5 above)
Comment (rithvikrao) [reply public/userview.html]: done

File: public/userview.html (hunk 6 above)
Comment (rithvikrao) [reply public/userview.html]: done

**Review by akainth015 — APPROVED** [review]
Awesome work!

**Pull request merged**